hsapp/flags.go

```diff
@@ -11,26 +11,75 @@ import (
 // Flags specifies app flags
 type Flags struct {
 	optionalConfigPath *string
-	bgColor            *uint
+	bgColor            *string
 	logFile            *string
 }
 
+// parse all of the command line args
 func (a *App) parseArgs() {
-	configDescr := fmt.Sprintf("specify a custom config path. Default is: %s", hsconfig.GetConfigPath())
-	a.Flags.optionalConfigPath = flag.String("config", "", configDescr)
-	a.Flags.bgColor = flag.Uint("bgColor", hsconfig.DefaultBGColor, "custom background color")
-	a.Flags.logFile = flag.String("log", "", "path to the output log file")
-	showHelp := flag.Bool("h", false, "Show help")
+	a.parseConfigArgs()
+	a.parseLogFileArgs()
+	a.parseBackgroundColorArgs()
+
+	// help args need to be parsed last, so that all other args
+	// can be parsed before a possible os.Exit() invoked by `-h` or `--help`
+	//
+	// otherwise, other flags will not be printed in usage string!
+	a.parseHelpArgs()
+}
+
+func (a *App) parseHelpArgs() {
+	const (
+		short    = "h"
+		long     = "help"
+		fmtUsage = "usage: %s [<flags>]\n\nFlags:\n"
+	)
+
+	// we will use a single variable for both short and long flags
+	var showHelp bool
+
+	flag.BoolVar(&showHelp, long, false, "Show help")
+	flag.BoolVar(&showHelp, short, false, "Show help (shorthand)")
 
 	flag.Usage = func() {
-		fmt.Printf("usage: %s [<flags>]\n\nFlags:\n", os.Args[0])
+		fmt.Printf(fmtUsage, os.Args[0])
 		flag.PrintDefaults()
 	}
 
 	flag.Parse()
 
-	if *showHelp {
+	if showHelp {
 		flag.Usage()
-		os.Exit(0)
+		os.Exit(0) // this is dangerous, forces us to parse the help flags last
```

Comment:
Bubble this up and let the caller's caller control the state.

Reply:
Handled this in 170ea91

Also moved `flag.Parse()` out of the `parseHelpArgs` method.

I had to add a private boolean member `app.showUsage`. this is what gets written to when `flag.Parse()` is finally called.